adapters/webhdfs-adapter/src/main/java/com/connexta/replication/adapters/webhdfs/WebHdfsNodeAdapter.java

```diff
@@ -290,9 +302,28 @@ boolean writeFileToLocation(CreateStorageRequest createStorageRequest, String lo
     return sendHttpRequest(httpPut, responseHandler);
   }
 
+  /**
+   * Converts an {@link UpdateStorageRequest} to a {@link CreateStorageRequest} by simply extracting
+   * the contained list of {@link Resource} and creating a new object from it.
+   *
+   * @param updateStorageRequest - the {@link UpdateStorageRequest} to convert
+   * @return The new {@link CreateStorageRequest} with the same {@link Resource}s
+   */
+  private CreateStorageRequest convertToCreateStorageRequest(
+      UpdateStorageRequest updateStorageRequest) {
+    if (updateStorageRequest.getResources().isEmpty()
+        || updateStorageRequest.getResources().get(0) == null) {
+      throw new ReplicationException(
```

Comment:
I would suggest catching this `ReplicationException` in your `updateResource(...)` method, so that you can print something using the `Logger` and return `false` from that method (similar to how `createResource(...)` is setup).